index.js

```diff
@@ -133,6 +138,58 @@ Shopify.prototype.request = function request(url, method, key, params) {
   });
 };
 
+/**
+ * Updates GraphQL API call limits.
+ *
+ * @param {String} throttleStatus The throttle status returned in the GraphQL response, under "extensions"
+ * @private
+ */
+Shopify.prototype.updateGraphqlLimits = function updateGraphqlLimits(throttleStatus) {
+  if (!throttleStatus) return;
+
+  const callGraphqlLimits = this.callGraphqlLimits;
+
+  callGraphqlLimits.remaining = throttleStatus.currentlyAvailable;
+  callGraphqlLimits.current = throttleStatus.maximumAvailable - throttleStatus.currentlyAvailable;
+  callGraphqlLimits.max = throttleStatus.maximumAvailable;
+
+  this.emit('callGraphqlLimits', callGraphqlLimits);
+};
+
+/**
+ * Sends a request to the Shopify GraphQL API endpoint.
+ *
+ * @param {Object} [data] Request body
+ * @return {Promise}
+ * @public
+ */
+Shopify.prototype.graphql = function graphql(data) {
+  const url = assign({ path: '/admin/api/graphql.json' }, this.baseUrl);
+  const options = assign({
+    headers: {
+      'User-Agent': `${pkg.name}/${pkg.version}`,
+      'Content-Type': 'application/graphql'
+    },
+    timeout: this.options.timeout,
+    json: false,
+    retries: 0,
+    method: 'POST',
+    body: data
+  }, url);
+
+  if (this.options.accessToken) {
+    options.headers['X-Shopify-Access-Token'] = this.options.accessToken;
+  }
+
+  return got(options).then(res => {
+    const body = res.body ? JSON.parse(res.body) : {};
+    this.updateGraphqlLimits(body.extensions && body.extensions.cost && body.extensions.cost.throttleStatus);
```

Comment:
I think this makes the

```js
if (!throttleStatus) return;
```

check above redundant? I would just call `this.updateGraphqlLimits()` if `throttleStatus` exists.

Reply:
Right